lib/HttpClient.js

```diff
@@ -667,28 +674,54 @@ HttpClient.prototype.request = function request(opts, cb) {
     opts.audit = this.audit;
 
     if (opts.retry === false) {
-        rawRequest(opts, cb);
+        rawRequest(opts, {}, cb);
         return;
     }
 
     var call;
     var retry = cloneRetryOptions(opts.retry);
-
+    var retryInfo = {
+        retries: [],
+        // attempts start at 1, and will increment on per retry
+        attempts: 1
+    };
+    // set keep alive only when retries are enabled
     opts._keep_alive = this._keep_alive;
-    call = backoff.call(rawRequest, opts, cb);
-    call.setStrategy(new backoff.ExponentialStrategy({
-        initialDelay: retry.minTimeout,
-        maxDelay: retry.maxTimeout
-    }));
-    call.failAfter(retry.retries);
-    call.on('backoff', this.emit.bind(this, 'attempt'));
-
-    call.start();
+
+    if (opts.exponentialBackoff === false) {
+        // if no backoff, we have to keep track of num retries ourselves instead
+        // of relying on backoff module.
+        rawRequest(opts, retryInfo, function retryRawRequest(err, req) {
+            if (err && retryInfo.attempts <= retry.retries) {
+                retryInfo.attempts++;
+                return rawRequest(opts, retryInfo, retryRawRequest);
+            }
+            return cb(err, req);
+        });
+    } else {
+        // if backoff is specified (default behavior) then use backoff module
+        // to keep track of retry count.
+        call = backoff.call(rawRequest, opts, retryInfo, cb);
+        call.setStrategy(new backoff.ExponentialStrategy({
+            initialDelay: retry.minTimeout,
+            maxDelay: retry.maxTimeout
+        }));
+        call.failAfter(retry.retries);
+        // TODO: is this emitted event actually useful? if you have multiple
+        // inflight requests, how do you differentiate them?
+        call.on('backoff', this.emit.bind(this, 'attempt'));
```

Comment:
Can we avoid using bind with `self`? Bind is slower.

Reply:
This is actually existing code, but I can turn this into a closure if you prefer.

Reply:
Touche, the event is from existing code, but I can move that into the same listener.